lib/packages/core/src/api/factories/account/setRewardRecipient.ts

```diff
@@ -32,13 +32,20 @@ export const setRewardRecipient = (service: BurstService): (
         deadline: number,
     ): Promise<TransactionId> => {
 
-        let parameters = {
+        const parameters = {
             recipient,
             deadline: 1440,
             feeNQT: convertNumberToNQTString(parseFloat(feeNQT)),
             publicKey: senderPublicKey
         };
-        const {unsignedTransactionBytes: unsignedHexMessage} = await service.send<TransactionResponse>('setRewardRecipient', parameters);
+        const {
```

Comment:
Humm. You don't need this. If brs returns an error it is automatically mapped into HttpError.  Try catch the Api method in Angular should do the trick. The catch block will receive the HttpError instance with responded error message

Reply:
Issue is this is not being considered an error (response is 200, just the payload is different). Also this API sometimes returns `errorDescription` and other times `error`. So wrapping this in a try/catch won't help. I suppose we could modify `service.send` to better throw errors if it finds `error` or `errorDescription` in the response, but that could break other things.

Reply:
Good call - the issue here is BRS returns `error` sometimes and `errorDescription` other times. I updated `burstService.ts` to support either of these scenarios.